cli/cmd/generate_azure.go

```diff
@@ -0,0 +1,737 @@
+			},
+			Response: &answer,
+		}); err != nil {
+			return err
+		}
+
+		// Based on response, prompt for actions
+		switch answer {
+		case AzureUserIntegrationNames:
+			if err := promptAzureIntegrationNameQuestions(config); err != nil {
+				return err
+			}
+		case AzureExistingStorageAcount:
+			if err := promptAzureStorageAccountQuestions(config); err != nil {
+				return err
+			}
+		case AzureSubscriptions:
+			if err := promptAzureSubscriptionQuestions(config); err != nil {
+				return err
+			}
+		case AzureManagmentGroup:
+			if err := promptAzureManagementGroupQuestions(config); err != nil {
+				return err
+			}
+		case AdvancedAdIntegration:
+			if err := promptAzureAdIntegrationQuestions(config); err != nil {
+				return err
+			}
+		case AzureRegionStorage:
+			if err := promptCustomizeAzureLoggingRegion(config); err != nil {
+				return err
+			}
+		case AzureAdvancedOptLocation:
+			if err := promptCustomizeAzureOutputLocation(extraState); err != nil {
+				return err
+			}
+			return nil
+		}
+
+		// Re-prompt if not done
+		innerAskAgain := true
+		if answer == AzureAdvancedOptDone {
+			innerAskAgain = false
+		}
+
+		if err := SurveyQuestionInteractiveOnly(SurveyQuestionWithValidationArgs{
+			Checks:   []*bool{&innerAskAgain},
+			Prompt:   &survey.Confirm{Message: QuestionAzureAnotherAdvancedOpt, Default: false},
+			Response: &innerAskAgain,
+		}); err != nil {
+			return err
+		}
+
+		if !innerAskAgain {
+			answer = AzureAdvancedOptDone
+		}
+	}
+
+	return nil
+}
+
+func azureConfigIsEmpty(config *azure.GenerateAzureTfConfigurationArgs) bool {
+	return !config.Config && !config.ActivityLog
+}
+
+func allSubscriptionsDisabled(config *azure.GenerateAzureTfConfigurationArgs) *bool {
+	allSubscriptionsDisabled := !config.AllSubscriptions
+	return &allSubscriptionsDisabled
+}
+
+func writeAzureGenerationArgsCache(config *azure.GenerateAzureTfConfigurationArgs) {
+	if !azureConfigIsEmpty(config) {
+		cli.WriteAssetToCache(CachedAzureAssetIacParams, time.Now().Add(time.Hour*1), config)
+	}
+}
+
+// entry point for launching a survey to build out the required Azure generation parameters
+func promptAzureGenerate(config *azure.GenerateAzureTfConfigurationArgs, extraState *AzureGenerateCommandExtraState) error {
+
+	// Cache for later use if generation is abandoned and in interactive mode
+	if cli.InteractiveMode() {
+		defer writeAzureGenerationArgsCache(config)
+		defer extraState.writeCache()
+	}
+	// These are the core questions that should be asked.
+	if err := SurveyMultipleQuestionWithValidation(
+		[]SurveyQuestionWithValidationArgs{
+			{
+				Prompt:   &survey.Confirm{Message: QuestionAzureEnableConfig, Default: true},
+				Response: &config.Config,
+			},
+			{
+				Prompt:   &survey.Confirm{Message: QuestionEnableActivityLog, Default: true},
+				Response: &config.ActivityLog,
+			},
+			{
+				Prompt:   &survey.Confirm{Message: QuestionEnableAdIntegration, Default: true},
+				Response: &config.CreateAdIntegration,
```

Comment:
Do we want these to default to true? 🤔

Reply:
I had these as true as these are the default values from the original terraform example from Azure, that is create Config, Activity Log and Active Directory integrations, so have kept these as they are

Reply:
I'm just thinking about consistency between clouds. I believe AWS and GCP both default all settings to false, to begin with.